cypress/deployed-and-local/integration/irsSuperUser/irs-super-user-can-download-stin.cy.ts

```diff
@@ -0,0 +1,79 @@
+import { ROLES } from '../../../../shared/src/business/entities/EntityConstants';
+import { getCypressEnv } from '../../../helpers/env/cypressEnvironment';
+import { loginAsPetitioner } from '../../../helpers/authentication/login-as-helpers';
+import { petitionerCreatesElectronicCase } from '../../../helpers/fileAPetition/petitioner-creates-electronic-case';
+import { petitionsClerkQcsAndServesElectronicCase } from '../../../helpers/documentQC/petitions-clerk-qcs-and-serves-electronic-case';
+
+describe('irs superuser integration', () => {
+  //if local or prod, skip this test
+  if (getCypressEnv().env === 'local' || getCypressEnv().env === 'prod') return;
+
+  const password = getCypressEnv().defaultAccountPass;
+  const userName = 'cypress_test_account_irs_super_user@example.com';
+
+  beforeEach(() => {
+    Cypress.session.clearCurrentSessionData();
+    cy.task('deleteAllIrsCypressTestAccounts');
+  });
+
+  after(() => {
+    cy.task('deleteAllIrsCypressTestAccounts');
+  });
+
+  it('should let an irs superuser view the reconciliation report and download a STIN', () => {
+    cy.task('createAccount', {
+      irsEnv: true,
+      password,
+      role: ROLES.irsSuperuser,
+      userName,
+    });
+
+    cy.task('getBearerToken', { isIrsEnv: true, password, userName }).then(
+      idToken => {
+        loginAsPetitioner();
+        petitionerCreatesElectronicCase().then(docketNumber => {
+          petitionsClerkQcsAndServesElectronicCase(docketNumber);
+          verifyReconciliationReportAndStinUrl(idToken as string);
+        });
+      },
+    );
+  });
+});
+
+/**
+ *
+ * This function asserts that the IRS Reconciliation report is accessible and that it contains
+ * at least one downloadable STIN record. Method fails test if either of these conditions is
+ * not met.
+ *
+ * @param bearerToken bearer token for IRS superuser session (see tasks::getBearerToken)
+ */
+const verifyReconciliationReportAndStinUrl = (bearerToken: string) => {
+  const url = `https://api-${Cypress.env('DEPLOYING_COLOR')}.${getCypressEnv().env}.ef-cms.ustaxcourt.gov/v2/reconciliation-report/today`;
```

Comment:
I'm curious why we need to hit the reconciliation report endpoint here. Would it be possible to intercept the serve case response instead to get the `docketEntryId` for the stin so the test doesn't have to do more work?

Reply:
We do this because it roughly mirrors what the IRS does - first they trawl through the reconciliation report to get a listing of docket entries, and with that list they download some subset of that list.   So in that respect I feel the extra work in the test is worth it.